src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final double MAX_VALUE_PROPERTIES = 50;
+    public static final String DEFAULT_COLOR = "white";
+    public static final double DEFAULT_RADIUS = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private Figure[] figures = new Figure [] {
+            new Circle(colorSupplier.getRandomColor(), random.nextDouble(MAX_VALUE_PROPERTIES)),
+            new Square(random.nextDouble(MAX_VALUE_PROPERTIES), colorSupplier.getRandomColor()),
+            new RightTriangle(random.nextDouble(MAX_VALUE_PROPERTIES),
```

Comment:
Interesting idea, but it won't give truly random figures. They are generated once per instance lifetime, we need to generate new figures per each method call, so we should move it there. Better to generate in each case of switch statement separately than keep them in array (all generated figures but one won't be used, it's not efficient)